File: src/main/old/cc/sukazyo/cono/morny/data/NbnhhshQuery.java
package cc.sukazyo.cono.morny.data;

import java.io.IOException;

import com.google.gson.Gson;

import cc.sukazyo.cono.morny.util.OkHttpPublic.MediaTypes;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;
import okhttp3.ResponseBody;

public class NbnhhshQuery {
	
	public static class Word {
		public String name;
		public String[] trans;
		public String[] inputting;
	}
	
	public static class GuessResult {
		public Word[] words;
	}
	
	public record GuessReq (String text) {}
	
	public static final String API_URL = "https://lab.magiconch.com/api/nbnhhsh/";
	public static final String API_GUESS_METHOD = "guess/";
	
	private static final OkHttpClient httpClient = new OkHttpClient();
	
	public static GuessResult sendGuess (String text) throws IOException {
		final String reqJsonText = new Gson().toJson(new GuessReq(text));
		Request request = new Request.Builder()
				.url(API_URL + API_GUESS_METHOD)
				.post(RequestBody.create(reqJsonText, MediaTypes.JSON))
				.build();
		try (Response response = httpClient.newCall(request).execute()) {
			final ResponseBody body = response.body();
			if (body == null) throw new IOException("Null body.");
			final String x = "{ \"words\": " + body.string() + " }";
			return new Gson().fromJson(x, GuessResult.class);
		}
	}
	
}

File: src/main/scala/cc/sukazyo/cono/morny/extra/NbnhhshQuery.scala
package cc.sukazyo.cono.morny.extra

import cc.sukazyo.cono.morny.util.SttpPublic.mornyBasicRequest
import com.google.gson.Gson
import sttp.client3.{asString, HttpError, SttpClientException, UriContext}
import sttp.client3.okhttp.OkHttpSyncBackend
import sttp.model.MediaType

object NbnhhshQuery {
	
	case class Word (
		name: String,
		trans: Array[String] = Array.empty,
		inputting: Array[String] = Array.empty
	)
	case class GuessResult (words: Array[Word])
	
	private val API_URL = uri"https://lab.magiconch.com/api/nbnhhsh/"
	private val API_GUESS_METHOD = uri"$API_URL/guess/"
	
	private val httpClient = OkHttpSyncBackend()
	
	@throws[HttpError[_]|SttpClientException]
	def sendGuess (text: String): GuessResult = {
		case class GuessRequest (text: String)
		val http = mornyBasicRequest
			.body(Gson().toJson(GuessRequest(text))).contentType(MediaType.ApplicationJson)
			.post(API_GUESS_METHOD)
			.response(asString.getRight)
			.send(httpClient)
		GuessResult(Gson().fromJson(http.body, classOf[Array[Word]]))
	}
	
}
